metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/CpuMeter.java

```diff
@@ -18,68 +18,126 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.lang.management.ManagementFactory;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
 
 import org.apache.commons.io.FileUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.netflix.spectator.api.BasicTag;
 import com.netflix.spectator.api.Id;
 import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Tag;
 
 public class CpuMeter {
   private static final Logger LOGGER = LoggerFactory.getLogger(CpuMeter.class);
 
+  public static final String STATISTIC = "statistic";
+
+  public static final Tag TAG_All = new BasicTag(STATISTIC, "allProcess");
+
+  public static final Tag TAG_CURRENT = new BasicTag(STATISTIC, "currentProcess");
+
   private double rate;
 
+  private double processRate;
+
   private long lastTotalTime;
 
   private long lastIdleTime;
 
+  private long lastProcessTime;
+
   private int cpuNum;
 
-  private Id id;
+  // process id
+  private String pid;
+
+  // allProcess
+  private Id AId;
+
+  // currentProcess
+  private Id CId;
 
   public CpuMeter(Id id) {
-    this.id = id;
+    this.AId = id.withTag(TAG_All);
+    this.CId = id.withTag(TAG_CURRENT);
+
+    this.pid = getCurrentPid();
     this.cpuNum = Runtime.getRuntime().availableProcessors();
     refreshCpu();
     rate = 0.0;
+    processRate = 0.0;
+  }
+
+  private String getCurrentPid() {
+    String name = ManagementFactory.getRuntimeMXBean().getName();
+    if (name.contains("@")) {
+      return name.substring(0, name.indexOf("@"));
+    }
+    LOGGER.warn("Failed to get current process id. {}", name);
```

Comment:
If PID is null, we should stop to calculate data. 
BTW, when will pid is null?

Reply:
do not know when will pid is null. I's impossible that the pid is null.

Reply:
If it is impossible, you can throw an IllegalStateException here.